core/store/caching_directory.hpp

```diff
@@ -0,0 +1,146 @@
+
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2022 ArangoDB GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Andrey Abramov
+////////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#include <absl/container/flat_hash_map.h>
+
+#include <shared_mutex>
+
+#include "store/directory.hpp"
+
+namespace iresearch {
+
+class MaxSizeAcceptor {
+ public:
+  explicit constexpr MaxSizeAcceptor(size_t max_size = 1024) noexcept
```

Comment:
NIT 
MaxCountAcceptor would be better. As it is count of files, not overall size